controllers/nicclusterpolicy_controller.go

```diff
@@ -249,20 +253,39 @@ func (r *NicClusterPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	}
 	r.stateManager = stateManager
 
-	builder := ctrl.NewControllerManagedBy(mgr).
+	ctl := ctrl.NewControllerManagedBy(mgr).
 		For(&mellanoxv1alpha1.NicClusterPolicy{}).
+		WithOptions(controller.Options{MaxConcurrentReconciles: 1}).
 		// Watch for changes to primary resource NicClusterPolicy
 		Watches(&source.Kind{Type: &mellanoxv1alpha1.NicClusterPolicy{}}, &handler.EnqueueRequestForObject{})
 
-	// Watch for changes to secondary resource DaemonSet and requeue the owner NicClusterPolicy
+	// we always add object with a same(static) key to the queue to reduce
+	// reconciliation count
+	qHandler := func(q workqueue.RateLimitingInterface) {
+		q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+			Namespace: consts.DefaultNamespace,
+			Name:      consts.NicClusterPolicyResourceName,
+		}})
+	}
+
+	updateEnqueue := handler.Funcs{
+		UpdateFunc: func(e event.UpdateEvent, q workqueue.RateLimitingInterface) {
```

Comment:
are you planning to also handle delete ? (post 1.4 ?) should we have an issue on it ?

Reply:
It will be handled in a separate PR

Reply:
it will be handled in a separate PR